src/solver/hydro/management/HydroErrorsCollector.cpp

```diff
@@ -0,0 +1,36 @@
+#include "antares/solver/hydro/management/HydroErrorsCollector.h"
+
+#include <yuni/core/logs.h>
+
+#include "antares/antares/fatal-error.h"
+
+namespace Antares
+{
+void HydroErrorsCollector::Collect(const std::string& area_name, const std::string& message)
+{
+    logs.error() << "In Area " << area_name << " " << message;
+    area_errors_counter_[area_name]++;
+    errors_limit_reached_ = area_errors_counter_[area_name] > 10;
+    stop_ = true;
+
+    if (errors_limit_reached_)
+    {
+        throw FatalError("Hydro validation has failed !");
+    }
```

Comment:
I think it would be best to have this behavior instead
```cpp
if (!errors_limit_reached_) {
    logs.error() << "...";
}
```
There's no need to `throw`, let higher level functions handle that. Otherwise the responsability of `HydroErrorsCollector` becomes unclear, we might even have cases where 9 errors are considered valid, but 11 are considered invalid, etc.